.github/workflows/check.yml

```
@@ -32,44 +31,78 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        if [[ "$RUNNER_OS" == "Linux" ]]; then
+             sudo apt-get update -y && sudo apt-get install fish csh
+        elif [[ "$RUNNER_OS" == "Windows"]]; then
+             if [ "${{ matrix.py }}" == "2.7" || "${{ matrix.py }}" == "pypy2" ]; then
+                 choco install vcpython27 --yes -f
+             fi
+        elif [[ "$RUNNER_OS" == "macOS" ]]; then
+             brew update -vvv && brew install fish tcsh
+        fi
```

You could also try actions level conditionals:
```suggestion
        ${{ runner.os == "Linux" && "sudo apt-get update -y && sudo apt-get install fish csh" }}
        ${{ runner.os == "macOS" && "brew update -vvv && brew install fish tcsh" }}
        ${{ runner.os == "Windows" && ( matrix.py == "2.7" || matrix.py == "pypy2" ) && "choco install vcpython27 --yes -f" }}
```

---

needed an || true at end, but this works 👍